### ISSUE TITLE
[Bug] Invalid URLs can be bookmarked and they crash the browser.

### ISSUE THREAD

**Issue by mhammond** [mozilla-mobile/fenix#14331]
## Steps to reproduce

* Open `https://1234.56.78.90` - as expected, Fenix will show a page saying it can't be loaded.
* Note that the "star" button is enabled - tap it to bookmark the page.
* Fenix crashes - logcat below

Note that application-services is extremely unlikely to ever allow invalid URLs to actually be created. So at a minimum, this exception should be caught. Ideally though, the UI affordances wouldn't allow the bookmark to be created in the first place as the user may be confused by the star never "lights up" to indicate the bookmark was actually created.

```
2020-08-28 10:24:26.781 8092-8127/org.mozilla.fenix.debug E/places::ffi: URL parse error: invalid IPv4 address
2020-08-28 10:24:26.816 8092-8092/org.mozilla.fenix.debug E/ExceptionHandler: Uncaught exception handled: 
    mozilla.appservices.places.UrlParseFailed: Error parsing URL: invalid IPv4 address
        at mozilla.appservices.places.RustError.intoException(RustError.kt:44)
        at mozilla.appservices.places.PlacesWriterConnection.doInsert(PlacesConnection.kt:1688)
        at mozilla.appservices.places.PlacesWriterConnection.createBookmarkItem(PlacesConnection.kt:652)
        at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$addItem$2.invokeSuspend(PlacesBookmarksStorage.kt:92)
        at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$addItem$2.invoke(Unknown Source:10)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:154)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at mozilla.components.browser.storage.sync.PlacesBookmarksStorage.addItem$suspendImpl(PlacesBookmarksStorage.kt:91)
        at mozilla.components.browser.storage.sync.PlacesBookmarksStorage.addItem(Unknown Source:0)
        at org.mozilla.fenix.browser.BaseBrowserFragment$bookmarkTapped$2.invokeSuspend(BaseBrowserFragment.kt:916)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
2020-08-28 10:24:26.821 8092-8092/org.mozilla.fenix.debug I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=mozilla.appservices.places.UrlParseFailed: Error parsing URL: invalid IPv4 address, breadcrumbs=...
2020-08-28 10:24:26.853 8092-8092/org.mozilla.fenix.debug I/mozac/CrashReporter: Showing prompt
2020-08-28 10:24:26.860 9386-9386/? I/eportingProces: Not late-enabling -Xcheck:jni (already on)
```

**Comment by liuche** [comment]
Easiest case would be, catch the exception, show a message saying this is a malformed url that can't be bookmarked.

**Comment by mariansima** [comment]
This issue was reproduced on Samsung A8 SM-A530F Android 9 Firefox 93.0a1 2021-08-29

**Comment by MozillaNoah** [comment]
Re-opening as I've been able to reproduce this on Nightly and it's trending on Sentry

**Comment by mhammond** [comment]
> it's trending on Sentry

Best @skhamis and I can tell, this particular issue reproduces on Nightly and Release - but for some reason doesn't have much presence on Sentry - maybe because people don't often try and bookmark invalid URLs? It's not clear if I was wrong when I closed it, or it has regressed since then - but it's not related to the big change we a-s just landed on Nightly. What is trending on Sentry is a new "invalid bookmark" error which *is* related to what we just landed, but unrelated to this specific issue. That will be fixed by https://github.com/mozilla/application-services/pull/4793
